utils/nctl/sh/assets/compile_node_launcher.sh

```
@@ -14,6 +14,13 @@ source "$NCTL"/sh/utils/main.sh
 pushd "$NCTL_CASPER_NODE_LAUNCHER_HOME" || \
     { echo "Could not find the casper-node-launcher repo - have you cloned it into your working directory?"; exit; }
 
+while getopts 'd' opt; do 
+    case $opt in
+        d ) export NCTL_COMPILE_TARGET="debug";;
+        * ) echo "nctl-compile only accepts optional flag -d to compile in debug mode."
```

It's not a `nctl-compile` it this case, right? It's rather `compile_node_launcher`.

I would suggest either of the two approaches:

1. use the generic `this script only accepts...` and then stop the script execution
2. (preferred by me) just silently ignore unknown parameter and continue

However, I'm not familiar with the requirements, so this is just a suggestion.

---

I agree that that this error handling might be a little ambitious for this part of the tool. I chose to implement option 2 and just ignore unknown parameters.